### PR TITLE
Improve Selenium tests

### PR DESCRIPTION
**Pull request by dcramer**
- Explicitly define phantomjs dependency
- Move selenium to fixtures (shared instance suite-wide)
- Simplify login hook

@getsentry/infrastructure @getsentry/ui

### CODE REVIEW

File: src/sentry/testutils/cases.py [hunk 1]
```diff
@@ -438,3 +440,27 @@ class CliTestCase(TestCase):
...
+class AcceptanceTestCase(LiveServerTestCase):
+    # Use class setup/teardown to hold Selenium and Percy state across all acceptance tests.
+    # For Selenium, this is done for performance to re-use the same browser across tests.
+    # For Percy, this is done to call initialize and then finalize at the very end after all tests.
+    #
+    # TODO: if acceptance tests are split across files, this will need to be refactored into a
+    # pytest plugin/fixture or something else that can manage global state.
+
+    @before
+    def clear_cookies(self):
```
Comment (dcramer) [inline src/sentry/testutils/cases.py]: couldn't figure out how to move this into a pytest fixture that only existed if browser was setup

File: src/sentry/testutils/cases.py [hunk 2]
```diff
@@ -438,3 +440,27 @@ class CliTestCase(TestCase):
...
         return self.runner.invoke(self.command, args, obj={})
+
+
+@pytest.mark.usefixtures('browser_class', 'percy_class')
+class AcceptanceTestCase(LiveServerTestCase):
+    # Use class setup/teardown to hold Selenium and Percy state across all acceptance tests.
+    # For Selenium, this is done for performance to re-use the same browser across tests.
+    # For Percy, this is done to call initialize and then finalize at the very end after all tests.
+    #
+    # TODO: if acceptance tests are split across files, this will need to be refactored into a
```
Comment (dcramer) [inline src/sentry/testutils/cases.py]: ill remove this comment

**Comment by fotinakis** [comment]
Niice! I tried to grok pytest fixtures but they were very weird. :) This makes them make much more sense. If you hit Approve on the Percy build, it should mark this PR green above.

**Comment by mattrobenolt** [comment]
For context, this is what happens in our production build:

```
    running [npm install --production --quiet]
    Error extracting archive
    Phantom installation failed { [Error: Command failed: tar jxf /usr/src/getsentry/src/sentry/node_modules/phantomjs-prebuilt/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
    tar (child): bzip2: Cannot exec: No such file or directory
    tar (child): Error is not recoverable: exiting now
    tar: Child returned status 2
    tar: Error is not recoverable: exiting now
    ]
      killed: false,
      code: 2,
      signal: null,
      cmd: 'tar jxf /usr/src/getsentry/src/sentry/node_modules/phantomjs-prebuilt/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2' } Error: Command failed: tar jxf /usr/src/getsentry/src/sentry/node_modules/phantomjs-prebuilt/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
    tar (child): bzip2: Cannot exec: No such file or directory
    tar (child): Error is not recoverable: exiting now
    tar: Child returned status 2
    tar: Error is not recoverable: exiting now
    
        at ChildProcess.exithandler (child_process.js:754:12)
        at ChildProcess.emit (events.js:110:17)
        at maybeClose (child_process.js:1019:16)
        at Socket. (child_process.js:1190:11)
        at Socket.emit (events.js:107:17)
        at Pipe.close (net.js:484:11)
    npm WARN optional dep failed, continuing fsevents@1.0.12
    npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
    node-pre-gyp ERR! Tried to download: https://node-zopfli.s3.amazonaws.com/Release/zopfli-v1.4.0-node-v14-linux-x64.tar.gz
    node-pre-gyp ERR! Pre-built binaries not found for node-zopfli@1.4.0 and node@0.12.13 (node-v14 ABI) (falling back to source compile with node-gyp)
    npm WARN deprecated graceful-fs@3.0.8: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
    npm ERR! Linux 3.13.0-68-generic
    npm ERR! argv "node" "/usr/local/bin/npm" "install" "--production" "--quiet"
    npm ERR! node v0.12.13
    npm ERR! npm  v2.15.4
    npm ERR! code ELIFECYCLE
    
    npm ERR! phantomjs-prebuilt@2.1.7 install: `node install.js`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the phantomjs-prebuilt@2.1.7 install script 'node install.js'.
    npm ERR! This is most likely a problem with the phantomjs-prebuilt package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     node install.js
    npm ERR! You can get information on how to open an issue for this project with:
    npm ERR!     npm bugs phantomjs-prebuilt
    npm ERR! Or if that isn't available, you can get their info via:
    npm ERR!
    npm ERR!     npm owner ls phantomjs-prebuilt
    npm ERR! There is likely additional logging output above.
    
    npm ERR! Please include the following file with any support request:
    npm ERR!     /usr/src/getsentry/src/sentry/npm-debug.log
    Traceback (most recent call last):
      File "/usr/src/getsentry/src/sentry/setup.py", line 329, in run
        self._build_static()
      File "/usr/src/getsentry/src/sentry/setup.py", line 389, in _build_static
        check_output(['npm', 'install', '--production', '--quiet'], cwd=work_path)
      File "/usr/local/lib/python2.7/subprocess.py", line 573, in check_output
        raise CalledProcessError(retcode, cmd, output=output)
    CalledProcessError: Command '['npm', 'install', '--production', '--quiet']' returned non-zero exit status 1
    unable to build Sentry's static assets!
    Hint: You might be running an invalid version of NPM.
    
    ----------------------------------------
Command "/usr/local/bin/python2 -c "import setuptools, tokenize;__file__='/usr/src/getsentry/src/sentry/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" develop --no-deps" failed with error code 1 in /usr/src/getsentry/src/sentry/
```

**Comment by mattrobenolt** [comment]
Turns out, this was failing for us because our production environment doesn't have `bzip2`, so `phantomjs-prebuilt` was failing to get installed. This also explains why I was having trouble reproducing this myself.

Fortunately, I was able to work around the issue because we don't even need to install this for our production builds, and I moved it to `devDependencies` with https://github.com/getsentry/sentry/commit/e30c39ebc2c1f6410b324d69968521ad149b5177.

**Pull request merged**